tests/numerics/wave_solver_hermite.cc

```diff
@@ -0,0 +1,425 @@
+  QGauss<dim - 1> face_quadrature(2 * regularity + 2);
+
+  Solution<dim>            wave(initial_time);
+  Solution_derivative<dim> wave_tdev(wave);
+
+  Vector<double> initial_conds_value(dof.n_dofs());
+  Vector<double> iniital_conds_tdev(dof.n_dofs());
+
+  Vector<double> sol_prev(dof.n_dofs());
+  Vector<double> sol_curr(dof.n_dofs());
+  Vector<double> sol_next(dof.n_dofs());
+
+  Vector<double> temp(dof.n_dofs());
+
+  SparsityPattern sp;
+  {
+    DynamicSparsityPattern dsp(dof.n_dofs());
+    DoFTools::make_sparsity_pattern(dof, dsp);
+    sp.copy_from(dsp);
+  }
+
+  SparseMatrix<double> mass, mass_solve, tstep_noninvert;
+  mass.reinit(sp), mass_solve.reinit(sp), tstep_noninvert.reinit(sp);
+
+  MatrixCreator::create_mass_matrix(mapping_h, dof, quadrature, mass);
+  MatrixCreator::create_laplace_matrix(mapping_h,
+                                       dof,
+                                       quadrature,
+                                       tstep_noninvert);
+  tstep_noninvert *= -0.5 * dt * dt;
+  tstep_noninvert.add(1.0, mass);
+
+  VectorTools::project_hermite(mapping_h,
+                               dof,
+                               constraints,
+                               quadrature,
+                               wave,
+                               sol_curr,
+                               false,
+                               face_quadrature,
+                               true);
+  VectorTools::project_hermite(mapping_h,
+                               dof,
+                               constraints,
+                               quadrature,
+                               wave_tdev,
+                               sol_prev,
+                               false,
+                               face_quadrature,
+                               true);
+
+  std::map<types::global_dof_index, double>           boundary_values;
+  std::map<types::boundary_id, const Function<dim> *> boundary_functions;
+
+  boundary_functions.emplace(std::make_pair(0U, &wave));
+  if (dim == 1)
+    boundary_functions.emplace(std::make_pair(1U, &wave));
+
+  std::vector<double> l2_errors;
+  double              max_error;
+  l2_errors.emplace_back(
+    l2_error<dim>(mapping_h, dof, quadrature, wave, sol_curr));
+  max_error = l2_errors[0];
+
+  // Make an initial time-step using
+  // u(t+dt) = u(t) + dt u_t(t) +
+  //              0.5 dt^2 M^(-1)A u(t) + O(dt^3, dt^2 dx^(p+1))
+  tstep_noninvert.vmult(temp, sol_curr);
+  mass.vmult(sol_next, sol_prev);
+  sol_next *= dt;
+  temp += sol_next;
+  sol_next = 0;
+
+  wave.update_time(dt);
+  VectorTools::project_hermite_boundary_values(
+    mapping_h,
+    dof,
+    boundary_functions,
+    face_quadrature,
+    VectorTools::HermiteBoundaryType::hermite_dirichlet,
+    boundary_values);
+
+  mass_solve.copy_from(mass);
+  MatrixTools::apply_boundary_values(
+    boundary_values, mass_solve, sol_next, temp, true);
+
+  SparseDirectUMFPACK mass_inv;
```

Comment:
I think you can compile deal.II without UMFPACK support. Either, we exclude the test if UMFPACK is missing or we use a simpler iterative deal.II iterative solver.